OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -146,7 +146,7 @@ public void Render(WorldRenderer wr)
 
 			Dirty.Clear();
 
-			foreach (var uv in wr.Viewport.VisibleCells.MapCoords)
+			foreach (var uv in wr.Viewport.VisibleCellsInsideBounds.MapCoords)
```

Comment:
Actually, this is a good idea.  I updated this and the terrain renderer, and then removed the magic Viewport behavior.